From 9e9e71bfa58bb4c6e3af834b3f19fabbdcd2ab1e Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 12 Aug 2014 23:58:29 -0400
Subject: [PATCH] Move WorldSaveEvent to proper location

calling CraftWorld.save() currently does not call WorldSaveEvent, and WorldSaveEvent could fire on worlds that have saving disabled.

New location will always fire during a world save and only during an actual save.
---
 src/main/java/net/minecraft/server/MinecraftServer.java | 4 ++--
 src/main/java/net/minecraft/server/WorldServer.java     | 1 +
 src/main/java/org/bukkit/craftbukkit/CraftServer.java   | 4 ++--
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index f56aac7..b7421d0 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -376,8 +376,8 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IMo
                     worldserver.save(true, (IProgressUpdate) null);
                     worldserver.saveLevel();
 
-                    WorldSaveEvent event = new WorldSaveEvent(worldserver.getWorld());
-                    this.server.getPluginManager().callEvent(event);
+                    // WorldSaveEvent event = new WorldSaveEvent(worldserver.getWorld()); // Spigot - moved to .save()
+                    // getPluginManager().callEvent(event); // Spigot - moved to .save()
                     // CraftBukkit end
                 }
             }
diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 8cbfa30..53e62f9 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -791,6 +791,7 @@ public class WorldServer extends World {
 
     public void save(boolean flag, IProgressUpdate iprogressupdate) throws ExceptionWorldConflict { // CraftBukkit - added throws
         if (this.chunkProvider.canSave()) {
+            org.bukkit.Bukkit.getPluginManager().callEvent(new org.bukkit.event.world.WorldSaveEvent(getWorld())); // Spigot
             if (iprogressupdate != null) {
                 iprogressupdate.a("Saving level");
             }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index cbbf1a3..f5cbb0c 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1084,8 +1084,8 @@ public final class CraftServer implements Server {
             try {
                 handle.save(true, null);
                 handle.saveLevel();
-                WorldSaveEvent event = new WorldSaveEvent(handle.getWorld());
-                getPluginManager().callEvent(event);
+                // WorldSaveEvent event = new WorldSaveEvent(handle.getWorld()); // Spigot - moved to .save()
+                // getPluginManager().callEvent(event); // Spigot - moved to .save()
             } catch (ExceptionWorldConflict ex) {
                 getLogger().log(Level.SEVERE, null, ex);
             }
-- 
1.9.1

